tee-worker/ts-tests/batch.test.ts

```diff
@@ -88,8 +98,13 @@ describeLitentry('Test Batch Utility', 0, (context) => {
             'identityManagement',
             ['IdentityCreated']
         );
-        const event_datas = await handleIdentityEvents(context, aesKey, resp_events, 'IdentityCreated');
-        for (let i = 0; i < event_datas.length; i++) {
+        const event_datas = (await handleIdentityEvents(
+            context,
+            aesKey,
+            resp_events,
+            'IdentityCreated'
+        )) as IdentityGenericEvent[];
```

Comment:
Why do we need the `as` here? :thinking:

Reply:
Then shouldn't we instead handle the `string` case in the caller and/or change the type of `assertIdentityCreated` accordingly? (bonus points for a more informative error type for the failure case as opposed to a naked `string`)

Reply:
What do you think about adding a `handleIdentityErrorEvent` specifically for handling error messages? I'm considering whether it would be too cumbersome to have `assertIdentityCreated` handle all the data.I can also separate the two return types and eliminate the need for as assertions.:p

Reply:
> What do you think about adding a handleIdentityErrorEvent specifically for handling error messages?

Could work; not sure how you'd want to integrate it with the batch case (e.g. if you have a batch where some transactions are supposed to succeed and some are supposed to fail, how would it look using `handleIdentityErrorEvent`?)

Reply:
Currently, what I'm thinking is to rewrite the `assertIdentityCreated/Verified/Removed`methods so that the obtained events are directly passed to them for processing, instead of handling them collectively in `handleIdentityEvents`. The advantages of doing this are:

1. Improved code readability and maintainability, as it separates the responsibilities.
2. We can use the `LitentryPrimitivesIdentity` interface when asserting, making it easier to perform checks.

Reply:
That sounds very reasonable :+1: